modules/apps/static/portal-osgi-web/portal-osgi-web-servlet-jsp-compiler/src/main/java/com/liferay/portal/osgi/web/servlet/jsp/compiler/internal/JSPServletFactoryImpl.java

```diff
@@ -53,6 +54,10 @@
 
 	@Activate
 	public void activate(BundleContext bundleContext) {
+		SystemProperties.set(
```

Comment:
@dantewang @yuhai 
Let's try to find another way to fix the threadlocal leaking. If we are sure which thread we are using, then we can remove it manually in our code. If we are not sure which thread, we have to go with system properties, please add a new property in system.properties.

Tina.

Reply:
@tinatian Got it.